rom/dev/src/flow/cold_reset/fmc_alias.rs

```diff
@@ -247,12 +251,63 @@ impl FmcAliasLayer {
 
         cprintln!(
             "[afmc] Image verified using Vendor ECC Key Index {}",
-            info.vendor_ecc_pub_key_idx
+            info.vendor_ecc_pub_key_idx,
         );
         report_boot_status(FmcAliasImageVerificationComplete.into());
         Ok(info)
     }
 
+    fn update_fuse_log(log_info: &ImageVerificationLogInfo) -> CaliptraResult<()> {
```

Comment:
Would it be better to have a fixed-size struct with the fuse log? Then it can be written to DCCM all at once.

Reply:
Right now we are logging all the fuses at once and the fixed-size struct can be used but at the same time that will take away the flexibility where we want to log a particular fuse values either multiple times or at different times in boot process.

Reply:
I was thinking something like this:

```rust
#[repr(C)]
struct FuseLogEntry {
  ...
}

#[repr(C)]
struct FuseLog {
  count: u32, // Number of entries
  entries: [FuseLogEntry; MAX_FUSE_LOG_ENTRIES];
}
```

I think this would make serializing/parsing the log simpler. We could just use the `zerocopy` crate to move this to/from DCCM.

Reply:
We did thought about having a count in place but decided not to for the following reason(s).
 
We have a entry id in each field and then entry_data. For each type of data there is separate entry ID. 0 is considered as invalid entry ID. Right now the parsing code is as simple as reading the entry and deserialize it in to 

```
pub struct FuseLogEntry {
     pub entry_id: u32,
     pub log_data: [u32; 1],
     pub reserved: [u32; 2],
}
```

Right now the log size is about 1k. Each entry is 16 bytes, which means that we can have at most 1024/16 = 64 entries. 
We can deserialize the entire 64 entries using zero copy in one shot and then parse them one by one till we do not hit a entry with a value of zero. In my opinion since the number of entries are fixed and little, we really do not need a count. 

Let me think/discuss this.

Reply:
Number of entries is fixed right now. But FMC/Runtime will need to support multiple different ROM images into the future. Depending on hardcoded values chosen by ROM is problematic because if ROM ever changes in future SoCs, higher level firmware gets very complicated.

Reply:
We are just following the PCR log pattern for this PR. Like the PCR, the number of fuses and the manifest data that we log is fixed and don't think that it will change. Having a count here would not help anyway. If we have a count in the structure we will still have to read the entire structure and then parse only the number of entries (still one by one). In my case where there is no count we still read the entire 64 entries and then parse the entries till we hit a invalid entry. Don't see much of a difference in two approaches.

Reply:
I don't understand the problem. `count` is only 4 bytes and then firmware doesn't have to do special-case parsing (fwiw, I would recommend the same thing for PCR log)

Reply:
Both the PCR and FUSE logs were basically designed to be no more than the snap-shot of the values. The entry-id of the log decides the location where the entry goes. This was done for the ease of parsing. But just to keep track of this discussion I have created a new issue and we will discuss this more. 

https://github.com/chipsalliance/caliptra-sw/issues/329